cmd/fakemachine/main.go

```diff
@@ -21,15 +21,35 @@ type Options struct {
 var options Options
 var parser = flags.NewParser(&options, flags.Default)
 
+func parseVolumeOptions(optionString string) (mpo []fakemachine.MountPointOption) {
+	mpo = make([]fakemachine.MountPointOption, 0)
+	opts := strings.Split(optionString, ",")
+	for _, opt := range opts {
+		parts := strings.SplitN(opt, ".", 2)
+		switch parts[0] {
+		case "virtfs":
+			mpo = append(mpo, fakemachine.VirtfsOption(parts[1]))
+		case "mount":
+			mpo = append(mpo, fakemachine.MountOption(parts[1]))
```

Comment:
I had to wrap my head a few times to work out how this worked ;). 

I'd change this to just add another fucntion (AddVolumeAtWithOptions or something with a nicer name :p) to pass on the options directly rather then parsing the string into a list of funcs which then get executed to add themselves to the mountpoint :)..

Bonus point for a creative solution though.